stubs/docker/docker/api/container.pyi

```diff
@@ -77,7 +78,7 @@ class ContainerApiMixin:
         since: Incomplete | None = None,
         follow: Incomplete | None = None,
         until: Incomplete | None = None,
-    ): ...
+    ) -> Generator[bytes, None, None] | bytes: ...
```

Comment:
It seems to me that it would make sense to use an overload based on `stream` and `demux`: 

* Return `CancellableStream` if `stream is `True`.
* Return `Generator[tuple[bytes | None, bytes | None], None, None]` if `demux` is `True`.
* Return `Generator[bytes, None, None]` otherwise.

Reply:
Thank you. I'll have a go at creating an overload - though this is something I have not done before.

I believe `demux` is not related to this function.

Reply:
@srittau - I have made an attempt to do this here (but not yet in the other file). I found it challenging to know what to do wrt e.g. an argument with a default followed by an argument without a default. I'd appreciate some tips / specific guidance on this change, and then I can apply it to the other file and hopefully to other changes I make to this repository.

Reply:
Arguments with defaults following arguments with defaults are a bit cumbersome, unfortunately. You need two overloads for that, one where all previous parameters are required, and one using the star import.

```python
    @overload
    def logs(
        self,
        container,
        stdout: bool,
        stderr: bool,
        *,
        stream: True,
        timestamps: bool = False,
        tail: str = "all",
        since: Incomplete | None = None,
        follow: Incomplete | None = None,
        until: Incomplete | None = None,
    ) -> CancellableStream: ...
    @overload
    def logs(
        self,
        container,
        stdout: bool,
        stderr: bool,
        stream: True,
        timestamps: bool = False,
        tail: str = "all",
        since: Incomplete | None = None,
        follow: Incomplete | None = None,
        until: Incomplete | None = None,
    ) -> CancellableStream: ...
    @overload
    def logs(
        self,
        container,
        stdout: bool = True,
        stderr: bool = True,
        stream: False = False,
        timestamps: bool = False,
        tail: str = "all",
        since: Incomplete | None = None,
        follow: Incomplete | None = None,
        until: Incomplete | None = None,
    ) -> bytes: ...
```

Reply:
Thank you!

Reply:
Also, I just noticed: `tail` should be annotated as `Literal["all"] | int`, but that may be outside the scope of this PR.

Reply:
> Also, I just noticed: tail should be annotated as Literal["all"] | int, but that may be outside the scope of this PR.

I've submitted that change in https://github.com/python/typeshed/pull/11906 and I will block this PR on that one to avoid conflicts.

Reply:
@srittau I'm working on the assumption that where you put e.g. `stream: True,`, that should be `stream: Literal[True],`